File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/push/RemoveAllPushChannelsForDevice.java
package com.pubnub.api.java.endpoints.push;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.models.consumer.push.PNPushRemoveAllChannelsResult;

public interface RemoveAllPushChannelsForDevice extends Endpoint<PNPushRemoveAllChannelsResult> {
    RemoveAllPushChannelsForDevice pushType(com.pubnub.api.enums.PNPushType pushType);

    RemoveAllPushChannelsForDevice deviceId(String deviceId);

    RemoveAllPushChannelsForDevice environment(com.pubnub.api.enums.PNPushEnvironment environment);

    RemoveAllPushChannelsForDevice topic(String topic);
}

File: pubnub-kotlin/src/main/kotlin/com/pubnub/api/endpoints/push/RemoveAllPushChannelsForDevice.kt
package com.pubnub.api.endpoints.push

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.push.PNPushRemoveAllChannelsResult

/**
 * @see [PubNub.removeAllPushNotificationsFromDeviceWithPushToken]
 */
interface RemoveAllPushChannelsForDevice : Endpoint<PNPushRemoveAllChannelsResult>